exasol_advanced_analytics_framework/udf_communication/peer_communicator/payload_message_sender.py

```diff
@@ -0,0 +1,59 @@
+from typing import List
+
+from exasol_advanced_analytics_framework.udf_communication import messages
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator.sender import Sender
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator.timer import Timer
+from exasol_advanced_analytics_framework.udf_communication.serialization import serialize_message
+from exasol_advanced_analytics_framework.udf_communication.socket_factory.abstract import Frame, Socket
+
+
+class PayloadMessageSender:
+    def __init__(self, message: messages.Payload,
+                 frames: List[Frame],
+                 retry_timer: Timer,
+                 abort_timer: Timer,
+                 sender: Sender,
+                 out_control_socket: Socket):
+        self._abort_timer = abort_timer
+        self._out_control_socket = out_control_socket
+        self._sender = sender
+        self._retry_timer = retry_timer
+        self._frames = frames
+        self._message = message
+        self._finished = False
+        self._send_payload()
+
+    def _send_payload(self):
+        self._sender.send_multipart(self._frames)
+
+    def try_send(self):
+        should_we_send_abort = self._should_we_send_abort()
```

Comment:
maybe better `shoud_we_abort_send`:
```suggestion
        should_we_abort_send = self._should_we_abort_send()
```